### DISCUSSION TITLE
How to intercept or other?

### DISCUSSION THREAD

**Discussion by donywan** [serverpod/serverpod#2767] [Q&A] [answered]
```
SystemUser.db
        .findFirstRow(session, where: (t) => t.username.equals(username) & 
        t.deleted.equals(Deleted.no.code));
```
I want to add `t.deleted.equals(Deleted.no.code)` to all query statements. How can I do this?

**Comment by SandPod** [comment, marked as answer]
Hey @donywan!

There currently is now way to add a global intercept or middleware to all database operations.

However, the `where` statement is simply a callback that takes a table and returns an expression.
This expression can be defined anywhere. So you could define your own method that returns an expression that you use in all your query statements.

```dart
SystemUser.db.findFirstRow(session, where: mySystemUserWhereExpression(t));
```

Where `mySystemUserWhereExpression` is a method that returns an expression.

**Reply by donywan** [reply]
This is also possible, but you need to write it like this every time you query. I hope you can add an interceptor later.
